app/main/views/login.py

```diff
@@ -146,9 +180,178 @@ def update_password(token):
                                **template_data), 400
 
 
+@main.route('/create-user/<string:encoded_token>', methods=["GET"])
+def create_user(encoded_token):
+    form = CreateUserForm()
+    template_data = main.config['BASE_TEMPLATE_DATA']
+
+    token = decode_invitation_token(encoded_token)
+
+    if token is None:
+        flash('token_invalid', 'error')
+        return render_template(
+            "auth/create-user.html",
+            form=form,
+            token=None,
+            email_address=None,
+            supplier=None,
+            **template_data), 400
+    else:
+        user = data_api_client.get_user(email_address=token.get("email_address"))
+
+        if user:
+            return render_template(
+                "auth/update-user.html",
+                user=user["users"],
+                email_address=token['email_address'],
+                supplier_name=token['supplier_name'],
+                token=encoded_token,
+                **template_data), 200
+        else:
+            return render_template(
+                "auth/create-user.html",
+                form=form,
+                email_address=token['email_address'],
+                supplier_name=token['supplier_name'],
+                token=encoded_token,
+                **template_data), 200
+
+
+@main.route('/update-user/<string:encoded_token>', methods=["POST"])
+def submit_update_user(encoded_token):
```

Comment:
The link prior to this blocks - so you don't see the create/update pages unless you're in a valid state.

Will add a check in here though in case some one just posts to this directly